lib/plugins/aws/invokeLocal/index.js

```diff
@@ -177,8 +189,145 @@ class AwsInvokeLocal {
+      let stdout = '';
+      docker.stdout.on('data', (buf) => { stdout += buf.toString(); });
+      docker.on('close', error => (error ? reject(error) : resolve(Boolean(stdout.trim()))));
+    });
+  }
+
+  pullDockerImage() {
+    const runtime = this.options.functionObj.runtime
+      || this.serverless.service.provider.runtime
+      || 'nodejs4.3';
+
+    this.serverless.cli.log('Downloading base docker image...');
+
+    return new BbPromise((resolve, reject) => {
+      const docker = spawn('docker', ['pull', `lambci/lambda:${runtime}`]);
+      docker.on('close', error => (error ? reject(error) : resolve()));
+    });
+  }
+
+  getLayerPaths() {
+    const layers = _.mapKeys(
+      this.serverless.service.layers,
+      (value, key) => this.provider.naming.getLambdaLayerLogicalId(key)
+    );
+
+    return Promise.all(
+      (this.options.functionObj.layers || this.serverless.service.provider.layers || [])
+        .map(layer => {
+          if (layer.Ref) {
+            return layers[layer.Ref].path;
+          }
+          const arnParts = layer.split(':');
+          const layerArn = arnParts.slice(0, -1).join(':');
+          const layerVersion = Number(arnParts.slice(-1)[0]);
+          const layerContentsPath = path.join(
+            '.serverless', 'layers', arnParts[6], arnParts[7]);
+          const layerContentsCachePath = path.join(
+            cachePath, 'layers', arnParts[6], arnParts[7]);
+          if (fs.existsSync(layerContentsPath)) {
+            return layerContentsPath;
+          }
+          let downloadPromise = Promise.resolve();
+          if (!fs.existsSync(layerContentsCachePath)) {
+            this.serverless.cli.log(`Downloading layer ${layer}...`);
+            mkdirp.sync(path.join(layerContentsCachePath));
+            downloadPromise = this.provider.request(
+              'Lambda', 'getLayerVersion', { LayerName: layerArn, VersionNumber: layerVersion })
+              .then(layerInfo => download(
+                layerInfo.Content.Location,
+                layerContentsPath,
+                { extract: true }));
+          }
+          return downloadPromise
+            .then(() => fse.copySync(layerContentsCachePath, layerContentsPath))
+            .then(() => layerContentsPath);
+        }));
+  }
+
+  buildDockerImage(layerPaths) {
+    const runtime = this.options.functionObj.runtime
+      || this.serverless.service.provider.runtime
+      || 'nodejs4.3';
+
+
+    const imageName = 'sls-docker';
+
+    return new BbPromise((resolve, reject) => {
+      let dockerfile = `FROM lambci/lambda:${runtime}`;
+      for (const layerPath of layerPaths) {
+        dockerfile += `\nADD --chown=sbx_user1051:495 ${layerPath} /opt`;
+      }
+      mkdirp.sync(path.join('.serverless', 'invokeLocal'));
+      const dockerfilePath = path.join('.serverless', 'invokeLocal', 'Dockerfile');
+      fs.writeFileSync(dockerfilePath, dockerfile);
+      this.serverless.cli.log('Building docker image...');
```

Comment:
Probably personal preference, but I'd vote for `Docker` vs. `docker` since it's a company / product name.

Reply:
Fair enough. When I think [Dd]ocker, I think `docker` bc it's `/usr/bin/docker` not `/usr/bin/Docker` 😜

Reply:
Make sense! 😊